### ISSUE TITLE
Temperatured softmax

### ISSUE THREAD

**Issue by smkia** [fchollet/keras#3092]
Any idea to add "temperatured softmax" activation layer to Keras?

**Comment by jmhessel** [comment]
This is quite simple to achieve. Basically, you can take your tensor that you want to compute the "temperatured" softmax of, divide it by the temperature, and then use the normal keras softmax.

You can achieve element-wise division using a lambda layer. Untested one-liner:

    model.add(Lambda(lambda x: x / temp))

for some temperature float temp.

**Comment by smkia** [comment]
Thanks @jmhessel , I like the idea of Lambda layer. My only concern is the gradient computation in the high-temperatured training phase. Are gradients same as normal softmax?

**Comment by jmhessel** [comment]
The forward pass is mathematically equivalent, so I don't see why the gradients wouldn't be the same too. Are you worried about numerical stability? What temperatures were you thinking of using?

**Comment by kechan** [comment]
> Thanks @jmhessel , I like the idea of Lambda layer. My only concern is the gradient computation in the high-temperatured training phase. Are gradients same as normal softmax?

Check out https://arxiv.org/pdf/1503.02531.pdf (Distillation) a scenario where T != 1 will matter.
